lib/beaker-pe/install/pe_utils.rb

```diff
@@ -1298,6 +1309,63 @@ def install_higgs( higgs_host = master )
           do_higgs_install higgs_host, options
         end
 
+        #Installs PE with a PE managed external postgres
+        def do_install_pe_with_pe_managed_external_postgres(hosts, opts, is_upgrade=false)
+          step "Setup tmp installer directory and pe.conf" do
+
+            prepare_hosts(hosts,opts)
+            register_feature_flags!(opts)
+            fetch_pe(hosts,opts)
+
+            [master, database, dashboard, pe_postgres].uniq.each do |host|
+              prepare_host_installer_options(host)
+
+              unless is_upgrade
+                setup_pe_conf(host, hosts, opts)
+              end
+            end
+          end
+
+          unless is_upgrade
+            step "Initial master install, expected to fail due to RBAC database not being initialized" do
+              begin
+                execute_installer_cmd(master, opts)
+              rescue => e
+                #The way the master fails to install includes this specific line as of PE 2018.1.x.
+                unless e.message =~ /The operation could not be completed because RBACs database has not been initialized/
+                  raise "Install on master failed in an unexpected manner"
+                end
+              end
+            end
+          end
+
+          step "Install/Upgrade postgres service on pe-postgres node" do
+            execute_installer_cmd(pe_postgres, opts)
+          end
+
+          step "Finish install/upgrade on infrastructure" do
+              [master, database, dashboard].uniq.each do |host|
+                execute_installer_cmd(host, opts)
+              end
+          end
+
+          step "Final puppet run on infrastructure + postgres node" do
+            [master, database, dashboard, pe_postgres].uniq.each do |host|
+              on host, 'puppet agent -t', :acceptable_exit_codes => [0,2]
+            end
+          end
+        end
+
+        #Method for handling install and upgrades of PE with postgres not on the same node as the puppetdb
+        def install_pe_with_pe_managed_external_postgres(hosts, opts)
+          #If we're not upgrading just do a standard install
+          if(original_pe_ver(hosts[0]) == hosts[0][:pe_ver])
```

Comment:
Since the only thing we are changing is the flag, I'm inclined to set the flag based on the conditional, and make a single call to do_install_pe_*

Reply:
Sounds good!